File: Reqnroll/Chapter13/13.4-TheEntitySelectorPattern/Before/WIMP.EntitySelectorSample.App/Models/Order.cs
namespace WIMP.EntitySelectorSample.App.Models;

public class Order(int orderNo, string customerName, string pizzaName, TimeSpan placingTime)
{
    public int OrderNo { get; set; } = orderNo;
    public string CustomerName { get; set; } = customerName;
    public string PizzaName { get; set; } = pizzaName;
    public TimeSpan PlacingTime { get; set; } = placingTime;
    public OrderStatus Status { get; set; } = OrderStatus.New;
}

File: CucumberJVM/Chapter13/13.4-EntitySelectorPattern/Before/src/main/java/com/wimp/app/models/Order.java
/*
 * NOTE: This application ("WIMP - Where Is My Pizza") is provided solely to
 * demonstrate the Behavior-Driven Development scenario automation patterns
 * described in the book "Effective Behavior-Driven Development" by
 * Gaspar Nagy and Seb Rose.
 *
 * It is NOT a complete or production-ready implementation. It deliberately
 * uses shortcuts and simplifications (e.g. authentication, data storage,
 * error handling, security) that are NOT suitable for a real application.
 * Do not use this code as a basis for production software.
 */

package com.wimp.app.models;

import java.time.LocalTime;

public class Order {
    private final int orderNo;
    private final String customerName;
    private final String pizzaName;
    private LocalTime placingTime;
    private OrderStatus status = OrderStatus.NEW;

    public Order(int orderNo, String customerName, String pizzaName, LocalTime placingTime) {
        this.orderNo = orderNo;
        this.customerName = customerName;
        this.pizzaName = pizzaName;
        this.placingTime = placingTime;
    }

    public int getOrderNo() {
        return orderNo;
    }

    public String getCustomerName() {
        return customerName;
    }

    public String getPizzaName() {
        return pizzaName;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }

    public LocalTime getPlacingTime() {
        return placingTime;
    }
}
